src/crafting.cpp

```diff
@@ -201,14 +205,18 @@ void finalize_recipes()
     }
 }
 
-bool player::crafting_allowed()
+bool player::crafting_allowed(const std::string rec_name) {
+    return crafting_allowed(*recipe_dict[rec_name]);
+}
```

Comment:
this overload allows the lua binding to `crafting_allowed()` to work with a string parameter. I wasn't able to figure out how/if it would be possible to inform the lua interface about the `recipe` type.